samples/add-feature-collection-layer-from-table/src/main/java/com/esri/arcgismaps/sample/addfeaturecollectionlayerfromtable/components/AddFeatureCollectionLayerFromTableViewModel.kt

```diff
@@ -0,0 +1,220 @@
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.data.FeatureCollection
+import com.arcgismaps.data.FeatureCollectionTable
+import com.arcgismaps.data.Field
+import com.arcgismaps.data.FieldType
+import com.arcgismaps.geometry.GeometryType
+import com.arcgismaps.geometry.Polygon
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.Polyline
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.FeatureCollectionLayer
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleRenderer
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel which creates three in-memory FeatureCollectionTables (point, polyline, polygon),
+ * adds features to them, creates a FeatureCollection and displays it on the map as a
+ * FeatureCollectionLayer.
+ */
+class AddFeatureCollectionLayerFromTableViewModel(app: Application) : AndroidViewModel(app) {
+
+    // ArcGIS map exposed for Compose screen.
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISOceans).apply {
+        initialViewpoint = Viewpoint(latitude = 8.849289, longitude = -79.497238, scale = 1e6)
+    }
+
+    // Message dialog view model for error handling
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        // Load the map and then build the feature collection layer
+        viewModelScope.launch {
+            createAndAddFeatureCollectionLayer()
+            arcGISMap.load()
+                .onFailure { messageDialogVM.showMessageDialog(it) }
+        }
+    }
+
+    private suspend fun createAndAddFeatureCollectionLayer() {
+
+            // Create each feature collection table
+            val pointsTable = createPointsCollectionTable()
+            val linesTable = createLinesCollectionTable()
+            val polygonsTable = createPolygonsCollectionTable()
+
+            // Create a FeatureCollection from the tables
+            val featureCollection = FeatureCollection(featureCollectionTables = listOf(pointsTable, linesTable, polygonsTable))
+
+            // Create a layer from the feature collection and add it to the map
+            val featureCollectionLayer = FeatureCollectionLayer(featureCollection = featureCollection)
+            arcGISMap.operationalLayers.add(featureCollectionLayer)
+    }
```

Comment:
indentation seems doubled here.
```suggestion
    private suspend fun createAndAddFeatureCollectionLayer() {

        // Create each feature collection table
        val pointsTable = createPointsCollectionTable()
        val linesTable = createLinesCollectionTable()
        val polygonsTable = createPolygonsCollectionTable()

        // Create a FeatureCollection from the tables
        val featureCollection = FeatureCollection(featureCollectionTables = listOf(pointsTable, linesTable, polygonsTable))

        // Create a layer from the feature collection and add it to the map
        val featureCollectionLayer = FeatureCollectionLayer(featureCollection = featureCollection)
        arcGISMap.operationalLayers.add(featureCollectionLayer)
    }
```